cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/aggregation/NGSIGenericAggregator.java

```diff
@@ -694,4 +694,22 @@ public void setEnableNameMappings(boolean enableNameMappings) {
      */
     public abstract void initialize(NGSIEvent cygnusEvent);
 
+    /**
+     * Returns the timestamp to use for the RECV_TIME_TS field. 
+     * @param cygnusEvent the event to produce the timestamp for.
+     * @return the timestamp value to use. Default milliseconds.
+     */
+    public long getRecvTimeTsValue(NGSIEvent cygnusEvent) {
+    	return cygnusEvent.getRecvTimeTs();
+    }
+    
+	/**
+	 * Returns a possible adapted value for attribute value. 
+	 * @param attrValue The input
+	 * @return the adapted output. Default unchanged, same object.
+	 */
+    public JsonElement adaptAttrValue(JsonElement attrValue) {
+    	// Default: No adaptation.
+    	return attrValue;
+    }
```

Comment:
This shouldn't be here. This data is very specific for this solution. This is a generic class which is used by many sinks.

Reply:
Perhaps, but the aggregate method is difficult to understand for non-insiders. By adding these idempotent method I was sure nothing to break and still able to inject a specific change. I do not see an alternative, even not after a fresh look.